build_library/qemu_template.sh

```diff
@@ -109,6 +123,29 @@ write_ssh_keys() {
     sed -e 's/^/ - /'
 }
 
+if [ -n "${SWTPM_DIR}" ]; then
+    mkdir -p "${SWTPM_DIR}"
+    if ! command -v swtpm >/dev/null; then
+        echo "$0: swtpm command not found!" >&2
```

Comment:
On Ubuntu, I needed to install some software packages and also initialize the swtpm:

```
sudo apt install swtpm swtpm-tools -y
sudo swtpm_setup --tpmstate $SWTPM_DIR \
    --create-ek-cert --create-platform-cert --lock-nvram --tpm2
```

Also, on Ubuntu, the only allowed socket path was in /run/swtpm/.

See: https://github.com/flatcar/Flatcar/issues/593#issuecomment-2025241150

Reply:
Inside the sdk_container docker process, I also tried to map the socket, but could not. Is there a way to map the socket?

From the docs, this should have worked, but it doesn't (no /run/swtpm/sock is present in the docker container):
```
./run_sdk_container -t -m /run/swtpm/sock:/run/swtpm/sock
```

Reply:
If we can test the TPM inside the container created by `./run_sdk_container`, there would be no need to break out back in the host OS to do the TPM testing.

Reply:
For Fedora it works directly, maybe we can find out when `swtpm_setup` needs to be run? Otherwise I would document this in the help output for the flag.

Reply:
> swtpm_setup --tpmstate

The procedure is documented here https://manpages.ubuntu.com/manpages/jammy/man8/swtpm_setup.8.html

Reply:
And here: https://github.com/stefanberger/swtpm/wiki/Certificates-created-by-swtpm_setup#tpm-2-certificates
I do not know how Fedora already created the tpm store, if you change the directory, it will work as well?

Reply:
For me this works with an empty folder (swtpm 0.8.1)

Maybe the config makes a difference?
```
$ cat /etc/swtpm_setup.conf 
# Program invoked for creating certificates
create_certs_tool= /usr/bin/swtpm_localca
create_certs_tool_config = /etc/swtpm-localca.conf
create_certs_tool_options = /etc/swtpm-localca.options
# Comma-separated list (no spaces) of PCR banks to activate by default
active_pcr_banks = sha256
```

Reply:
I've pushed a vague note in the help output. Since this may be different for the various distributions we probably can't cover all of them - some people might also want to run this on MacOS and I don't know how if the setup would be required there and how it would look.

Reply:
I think we can safely add this documentation on the flatcar-website, how to test the TPM functionality in isolation from the helper on Fedora and Ubuntu (this way, most of the generic OSses are covered).

Reply:
> For me this works with an empty folder (swtpm 0.8.1)
> 
> Maybe the config makes a difference?
> 
> ```
> $ cat /etc/swtpm_setup.conf 
> # Program invoked for creating certificates
> create_certs_tool= /usr/bin/swtpm_localca
> create_certs_tool_config = /etc/swtpm-localca.conf
> create_certs_tool_options = /etc/swtpm-localca.options
> # Comma-separated list (no spaces) of PCR banks to activate by default
> active_pcr_banks = sha256
> ```

On Ubuntu 22.04, swtpm 0.6.3:
```
cat /etc/swtpm_setup.conf
# Program invoked for creating certificates
create_certs_tool= /usr/lib/x86_64-linux-gnu/swtpm/swtpm-localca
create_certs_tool_config = /etc/swtpm-localca.conf
create_certs_tool_options = /etc/swtpm-localca.options
```